metal/cloud.go

```diff
@@ -29,8 +29,8 @@ const (
 type cloud struct {
 	client                      *packngo.Client
 	config                      Config
-	instances                   cloudprovider.InstancesV2
-	loadBalancer                cloudprovider.LoadBalancer
+	instances                   *instances
+	loadBalancer                *loadBalancers
```

Comment:
This fixes the issue where we were testing an interface type for nil to determine if the loadbalancer controller should be enabled or not.

Reply:
We actually had pointers to the concrete types in one of the PRs, but switched them to the interface types at the suggestion of @displagueto makes it easier to pass things in for testing and keep it standard.

Where do we do that testing?

Also, is it not legitimate to test if an interface type is nil?

Reply:
> Silly test, but look at https://go.dev/play/p/yC4BLAqvccn
> 
> I can legitimately test an interface type for nil.

Here is a better example of what I mean: https://go.dev/play/p/OkcY_Nvsc3H

Since we are setting the value with the returned value of a constructor, the type of the argument at runtime is not actually nil

Reply:
> We actually had pointers to the concrete types in one of the PRs, but switched them to the interface types at the suggestion of @displagueto makes it easier to pass things in for testing and keep it standard.
> 
> Where do we do that testing?
> 
> Also, is it not legitimate to test if an interface type is nil?

I didn't see that type of testing done anywhere, so I think it's probably best to use the pointers for now unless we have a real need for it later, in which case we can introduce additional logic to check for nil where needed.

Reply:
cc'ing @displague for visibilty

Reply:
If you both are ok with it, I have no issues. @displague want to comment?

Reply:
> Here is a better example of what I mean: https://go.dev/play/p/OkcY_Nvsc3H
> Since we are setting the value with the returned value of a constructor, the type of the argument at runtime is not actually nil

In your example, `f` actually is `nil`, as it never was set, but `h` returns from the constructor, so it is not `== nil`, but still will give an error if you try to call something on it. So basically, `== nil` does not catch all of the cases where we might have that issue. Is that it?

Do we actually trip this up somewhere in CPEM? I assume we do, or you wouldn't trouble yourself with this PR. Where is it?

Reply:
Got it. Yeah, then this is a much better way. Let's Gove @displague a chance to comment

Reply:
@deitch Can we go ahead and push forward with this? If needed we can always come back around and re-address the issue if needed.

Reply:
Yup, plenty of time for people to respond, and we are not yet cutting a release anyways.